### DISCUSSION TITLE
vm_memory_high_watermark is less than expected

### DISCUSSION THREAD

**Discussion by bdaoudtdc** [rabbitmq/cluster-operator#670] [Q&A] [answered]
## Describe the bug
I setup me resources as follows 
resources:
    requests:
      cpu: 1
      memory: 6Gi
    limits:
      cpu: 4
      memory: 6Gi

My  vm_memory_high_watermark is set to 0.7

I expected the watermark to be 0.7*6Gi  , but what I see in the logs is 

Memory high watermark set to 3440 MiB (3607772529 bytes) of 4915 MiB (5153960756 bytes) total

Not sure why the max is set to  4915 MiB and not to  6144 MiB

When I login to the container , I see the full amount that is assigned to the host
rabbitmq@rmq-qa-cluster-server-2:/etc$ free
              total        used        free      shared  buff/cache   available
Mem:       16247656     8008900     2975708      869608     5263048     7031112
Swap:             0           0           0

## Version and environment information
- RabbitMQ: [3.8.12]
- RabbitMQ Cluster Operator: [1.5.0]
- Kubernetes: [1.17.13]

**Comment by ChunyiLyu** [comment, marked as answer]
Hi @bdaoudtdc, I have converted the issue to a discussion since this is expected behavior and not a bug. A while back we added a feature to cluster operator so it provides headroom for the memory limit (see: https://github.com/rabbitmq/cluster-operator/blob/c813d56785460dd9ecef7a2423851c90afeeaf4e/internal/resource/configmap_test.go#L505-L516). This is to avoid having pods killed by k8s with OOM kill because RabbitMQ may temporary exceeds its high memory water mark. The headroom we've decided on is 20% with a maximum of 2GB and that's why the memory high watermark and total memory is different from what you've expected.
